mkopn/installer.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+# disable terminal receiver
+stty -cread
+
+# enable tracing
+set -eux
+
+# sanity check
+[ ${1} = /dev/mmcblk0p3 ] || exit 1
+[ ${2} = /dev/mmcblk0p4 ] || exit 1
+
+IMAGE_FILE=${3}
+
+# write root
+ROOT_IMAGE_OFFSET=$(printf "%d" "0x$(hexdump -s136 -n8 -v -e "8/1 \"%02x\"" ${IMAGE_FILE})")
+dd if=${IMAGE_FILE} bs=1024 skip=$((${ROOT_IMAGE_OFFSET} / 1024)) | gunzip -c | dd of=/dev/mmcblk0p3
```

Comment:
Did you check for the best block size (`bs`) parameter?

Reply:
No, but

1. this is non-trivial because of the gunzip in between (it may have side-effects on the block size) and
2. the root filesystem is aligned at every 1024 bytes therefore I may need to skip a integer number of blocks of 1024 bytes.

Reply:
I think, we should definitely try to at least optimize 3-4 times using grid search. This may come with extreme speedups.